flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MySqlToIcebergE2eITCase.java

```diff
@@ -0,0 +1,372 @@
+            recordsInIncrementalPhase = createChangesAndValidate(stat);
+        } catch (SQLException e) {
+            LOG.error("Update table for CDC failed.", e);
+            throw e;
+        }
+        List<String> recordsInSnapshotPhase =
+                new ArrayList<>(
+                        Arrays.asList(
+                                "102, Two, Bob, 1.703, white, {\"key2\": \"value2\"}, null, null, null, null, null, null, null, null, null, null",
+                                "103, Three, Cecily, 4.105, red, {\"key3\": \"value3\"}, null, null, null, null, null, null, null, null, null, null",
+                                "104, Four, Derrida, 1.857, white, {\"key4\": \"value4\"}, null, null, null, null, null, null, null, null, null, null",
+                                "105, Five, Evelyn, 5.211, red, {\"K\": \"V\", \"k\": \"v\"}, null, null, null, null, null, null, null, null, null, null",
+                                "106, Six, Fay, 9.813, null, null, null, null, null, null, null, null, null, null, null, null",
+                                "107, Seven, Grace, 5.125, null, null, null, null, null, null, null, null, null, null, null, null",
+                                "108, Eight, Hesse, 6.819, null, null, null, null, null, null, null, null, null, null, null, null",
+                                "109, Nine, IINA, 5.223, null, null, null, null, null, null, null, null, null, null, null, null",
+                                "110, Ten, Jukebox, 0.2, null, null, null, null, null, null, null, null, null, null, null, null",
+                                "111, Eleven, Kryo, 5.18, null, null, null, null, null, null, null, null, null, null, null, null",
+                                "112, Twelve, Lily, 2.14, null, null, null, null, null, null, null, null, null, null, null, null"));
+        recordsInSnapshotPhase.addAll(recordsInIncrementalPhase);
+        recordsInSnapshotPhase =
+                recordsInSnapshotPhase.stream().sorted().collect(Collectors.toList());
+        validateSinkResult(warehouse, database, "products", recordsInSnapshotPhase);
+    }
+
+    /**
+     * Basic Schema: id INTEGER NOT NULL, name VARCHAR(255) NOT NULL, description VARCHAR(512),
+     * weight FLOAT, enum_c enum('red', 'white'), json_c JSON.
+     */
+    private List<String> createChangesAndValidate(Statement stat) throws SQLException {
+        List<String> result = new ArrayList<>();
+        StringBuilder sqlFields = new StringBuilder();
+
+        // Add Column.
+        for (int addColumnRepeat = 0; addColumnRepeat < 10; addColumnRepeat++) {
+            stat.execute(
+                    String.format(
+                            "ALTER TABLE products ADD COLUMN point_c_%s VARCHAR(10);",
+                            addColumnRepeat));
+            sqlFields.append(", '1'");
+            StringBuilder resultFields = new StringBuilder();
+            for (int j = 0; j < 10; j++) {
```

Comment:
minor: we give meaningful variable name to avoid to use `i` and `j`, and the 10 is a magic number, we can define a constant with a meaningful name